decryptor/postgresql/pg_general_decryptor.go

```diff
@@ -476,15 +397,176 @@ func (decryptor *PgDecryptor) MatchZoneInBlock(block []byte) {
 
 // GetTagBeginLength returns begin tag length, depends on decryptor type
 func (decryptor *PgDecryptor) GetTagBeginLength() int {
-	return decryptor.pgDecryptor.GetTagBeginLength()
+	return decryptor.binaryDecryptor.GetTagBeginLength()
 }
 
 // GetZoneIDLength returns begin tag length, depends on decryptor type
 func (decryptor *PgDecryptor) GetZoneIDLength() int {
-	return decryptor.pgDecryptor.GetTagBeginLength()
+	return decryptor.binaryDecryptor.GetTagBeginLength()
 }
 
 // SetDataProcessor replace current with new processor
 func (decryptor *PgDecryptor) SetDataProcessor(processor base.DataProcessor) {
 	decryptor.dataProcessor = processor
 }
+
+// OnColumn handler which process column data on db response and try to decrypt/detect poison record
+func (decryptor *PgDecryptor) OnColumn(ctx context.Context, data []byte) (context.Context, []byte, error) {
+	logger := logging.GetLoggerFromContext(ctx)
+	span := trace.FromContext(ctx)
+	// try to skip small piece of data that can't be valuable for us
+	if (decryptor.IsWithZone() && len(data) >= zone.ZoneIDBlockLength) || len(data) >= base.GetMinAcraStructLength() {
```

Comment:
just IMHO.

For better readability, I suggest to change order:
- first, test that data len is too short and exit
- then, continue to parse AcraStruct assuming that data len is enough

This will change code structure from

```
if (data len is enough) {
 many many
 many many
 many many
 lines
}

"if data len is too short" condition branch, easy to miss
```

to

```
if data len is too short <- visible immediately
  return

// else 
many many 
many lines of code
```